### PR TITLE
Feature/refactor

### PR DESCRIPTION
**Pull request by Tony---Zhang**
1. Refactor code;
2. Update LCD display message;
3. Add debug macro to display useful message in console;
4. Fix some logic issues.
5. Format code.

### CODE REVIEW

File: src/BeerBrew/BeerBrew.ino [hunk 1]
```diff
@@ -7,462 +7,509 @@
...
+            check_result = check_temperature(tempvalue, (float) stage2_temp, HEATER_TEMPERATURE_OVERFLOW, true);
+            break;
+        case STAGE_BOIL:
+            display_current_temperature(tempvalue, stage3_temp);
+            display_current_time(lcdtime);
+            check_result = check_temperature(tempvalue, (float) stage3_temp, HEATER_TEMPERATURE_OVERFLOW, true);
+            break;
+        case STAGE_FERMERTATION:
+            display_current_temperature(tempvalue, stage4_temp);
+            sprintf(line, "Fermertation    ");
```
Comment (ripple0328) [inline src/BeerBrew/BeerBrew.ino]: fermentation

File: src/BeerBrew/BeerBrew.ino [hunk 2]
```diff
@@ -7,462 +7,509 @@
...
+#ifdef DEBUG
+    char log[64] = "";
+    sprintf(log, "Time: %02d:%02d:%02d\tpassed: %d", end_t.Hour, end_t.Minute, end_t.Second, *pass_time);
     Serial.println(log);
+#endif
 }
 
+void work() {
+    float tempvalue;
+    char lcdtime[17] = "";
```
Comment (ripple0328) [inline src/BeerBrew/BeerBrew.ino:402]: it will leave a ')' on the right side of the time
we could add a space between current time and left time

**Comment by ripple0328** [comment]
Great Refactor 👍

**Comment by Tony---Zhang** [comment]
Fixed issue.

**Pull request merged**